tests/topotests/multicast_pim6_dual_stack/test_multicast_pim6_dual_stack.py

```diff
@@ -0,0 +1,1952 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: ISC
+#
+# Copyright (c) 2023 by VMware, Inc. ("VMware")
+# Used Copyright (c) 2018 by Network Device Education Foundation,
+# Inc. ("NetDEF") in this file.
+#
+
+"""
+Following tests are covered to test multicast pim6 dual stack:
+
+1. Verify IPv4 and IPv6 mroute and upstream when mld and MLD join sent from same interface
+   of FRR1 and source present on same interface in FRR4
+2. Verify IPv4 and IPv6 mroute and upstream when mld and MLD join sent from different interface
+   of FRR1 and source present on different interface in FRR4
+3. Verify IPv4 and IPv6 mroute and upstream when mld and MLD join sent from same interface
+   of FRR1 and source present on same interface in FRR4
+4. Verify IPv4 and IPv6 mroute and upstream when mld and MLD join sent from same interface
+   of FRR1 and source present on same interface in FRR4
+5. Verify IPv4 and IPv6 mroute and upstream after removing/adding  IGMP and MLD config
+"""
+
+import os
+import sys
+import time
+import pytest
+import time
+import datetime
+
+# Save the Current Working Directory to find configuration files.
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, "../"))
+sys.path.append(os.path.join(CWD, "../lib/"))
+
+# Required to instantiate the topology builder class.
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib.topogen import Topogen, get_topogen
+
+from lib.common_config import (
+    start_topology,
+    write_test_header,
+    write_test_footer,
+    step,
+    reset_config_on_routers,
+    shutdown_bringup_interface,
+    required_linux_kernel_version,
+)
+from lib.pim import (
+    create_pim_config,
+    create_mld_config,
+    verify_mld_groups,
+    verify_mroutes,
+    verify_upstream_iif,
+    verify_pim_interface_traffic,
+    verify_sg_traffic,
+    McastTesterHelper,
+    create_igmp_config,
+    verify_igmp_groups,
+    verify_pim_rp_info,
+)
+from lib.topolog import logger
+from lib.topojson import build_config_from_json
+
+# Global variables
+GROUP_RANGE_1 = [
+    "ffaa::1/128",
+    "ffaa::2/128",
+    "ffaa::3/128",
+    "ffaa::4/128",
+    "ffaa::5/128",
+]
+MLD_JOIN_RANGE_1 = ["ffaa::1", "ffaa::2", "ffaa::3", "ffaa::4", "ffaa::5"]
+
+GROUP_RANGE_4 = [
+    "226.1.1.1/32",
+    "226.1.1.2/32",
+    "226.1.1.3/32",
+    "226.1.1.4/32",
+    "226.1.1.5/32",
+]
+IGMP_JOIN_RANGE_1 = ["226.1.1.1", "226.1.1.2", "226.1.1.3", "226.1.1.4", "226.1.1.5"]
+
+
+GROUP_RANGE_2 = [
+    "ffbb::1/128",
+    "ffbb::2/128",
+    "ffbb::3/128",
+    "ffbb::4/128",
+    "ffbb::5/128",
+]
+IGMP_JOIN_RANGE_2 = ["ffbb::1", "ffbb::2", "ffbb::3", "ffbb::4", "ffbb::5"]
```

Comment:
This one should be named as MLD_JOIN_RANGE.

Reply:
IGMP_JOIN_RANGE_2 not used, removed it.